src/main/kotlin/io/openfuture/state/controller/domain/dto/WalletDto.kt

```diff
@@ -0,0 +1,10 @@
+package io.openfuture.state.controller.domain.dto
+
+import java.time.LocalDateTime
+
+data class WalletDto(
```

Comment:
Wrong validation. When a user will send you `NULL` for `id` or `address` you will not show a proper message. The message would be from Kotlin type validation.

For all DTOs, Requests wrappers you always have to define all properties as nullable and annotate them with validation annotations like `@NotNull` and `@NotEmpty` and use `@Valid` in the controller.
If the validation passed, you can be sure fields are not null and a user will get a proper status code and the error message.